### PR TITLE
Improved material/cdk table integration

### PR DESCRIPTION
**Pull request by kadosh**
- Removed grid-table
- Updated references for material/cdk

### CODE REVIEW

File: src/lib/src/column-filter-dialog/column-filter-dialog.ts [hunk 1]
```diff
@@ -101,13 +106,14 @@ export class ColumnFilterDialogComponent implements AfterViewInit, OnInit {
...
 
     public togglePopover() {
-
-        if (ColumnFilterDialogComponent.prevPopover !== null && ColumnFilterDialogComponent.prevPopover !== this.popover) {
-            ColumnFilterDialogComponent.prevPopover.close();
-            ColumnFilterDialogComponent.prevPopover = null;
-        }
-
-        ColumnFilterDialogComponent.prevPopover = this.popover;
+        // console.log("was open", this.popover.isOpen());
```
Comment (geoperez) [inline src/lib/src/column-filter-dialog/column-filter-dialog.ts]: remove console.log

File: src/lib/src/grid/grid.spec.ts [hunk 2]
```diff
@@ -1,16 +1,82 @@
...
-    const mockJson = {
+    let simpleGridApp: SimpleGridApp;
+    let fixture: ComponentFixture<SimpleGridApp>;
+    let spy: jasmine.Spy;
+    let dataService: DataService;
+    let overlayContainerElement: HTMLElement;
+    let dir: { value: 'ltr' | 'rtl' };
+    let scrolledSubject = new Subject();
+
+    const mockJsonDefault = {
```
Comment (geoperez) [inline src/lib/src/grid/grid.spec.ts]: Maybe load Mocks from external file...

File: src/lib/src/grid/grid.spec.ts [hunk 3]
```diff
@@ -40,83 +106,320 @@ describe('Component: GridComponent', () => {
...
+
+function expectTextContent(el, text) {
+    if (el && el.textContent) {
+        expect(el.textContent.trim()).toBe(text);
+    } else {
+        fail(`Missing text content of ${text} in element ${el}`);
+    }
+}
+
+@Component({
```
Comment (geoperez) [inline src/lib/src/grid/grid.spec.ts]: Same as payloads, maybe load this component from external file

File: src/lib/src/grid/grid.ts [hunk 4]
```diff
@@ -184,13 +257,15 @@ export class GridComponent implements OnInit {
 
         const ngRequest = new Request(ngRequestOptions);
 
-        return this.http.request(ngRequest).map(response => {
+        const result = this.dataService.getData(ngRequest);
```
Comment (geoperez) [inline src/lib/src/grid/grid.ts:286]: How do you handle an error?

**Comment by kadosh** [comment]
So... how about Squash and merge? Anything you want to add?

**Pull request merged**